packages/driver-test-support/lib/e2e-suite.js

```diff
@@ -148,6 +149,11 @@ export function driverE2ETestSuite(DriverClass, defaultCaps = {}) {
 
     describe('session handling', function () {
       it('should handle idempotency while creating sessions', async function () {
+        if (util.compareVersions(process.version, '>=', '19.0.0')) {
+          // FIXME: Unfortunately the 'socket hang up' error is too hard to debug
```

Comment:
what is this about?

Reply:
if a test is breaking for node 19/20 maybe we should figure out what's wrong?

Reply:
I tried, but was not able to. If you can help, please do

Reply:
@mykola-mokhnach apply this

```diff
diff --git a/packages/driver-test-support/lib/e2e-suite.js b/packages/driver-test-support/lib/e2e-suite.js
index fcbb65ddf..650cc5197 100644
--- a/packages/driver-test-support/lib/e2e-suite.js
+++ b/packages/driver-test-support/lib/e2e-suite.js
@@ -5,6 +5,7 @@ import B from 'bluebird';
 import {TEST_HOST, getTestPort, createAppiumURL} from './helpers';
 import chai from 'chai';
 import sinon from 'sinon';
+import { Agent } from 'node:http';
 
 const should = chai.should();
 
@@ -148,9 +149,12 @@ export function driverE2ETestSuite(DriverClass, defaultCaps = {}) {
 
     describe('session handling', function () {
       it('should handle idempotency while creating sessions', async function () {
+
+        const httpAgent = new Agent({ keepAlive: true });
         const sessionIds = [];
         let times = 0;
         do {
+
           const {sessionId} = await startSession(
             {
               capabilities: {alwaysMatch: defaultCaps},
@@ -159,6 +163,7 @@ export function driverE2ETestSuite(DriverClass, defaultCaps = {}) {
               headers: {
                 'X-Idempotency-Key': '123456',
               },
+              httpAgent,
               // XXX: I'm not sure what these are, as they are not documented axios options,
               // nor are they mentioned in our source
               // @ts-expect-error
@@ -180,6 +185,7 @@ export function driverE2ETestSuite(DriverClass, defaultCaps = {}) {
       it('should handle idempotency while creating parallel sessions', async function () {
         const reqs = [];
         let times = 0;
+        const httpAgent = new Agent({ keepAlive: true });
         do {
           reqs.push(
             startSession(
@@ -189,6 +195,7 @@ export function driverE2ETestSuite(DriverClass, defaultCaps = {}) {
                 },
               },
               {
+                httpAgent,
                 headers: {
                   'X-Idempotency-Key': '12345',
                 },
```

Reply:
(see https://github.com/node-fetch/node-fetch/issues/1735)

Reply:
how much time have you spent debugging this @boneskull ?
I had an idea this is not as simple as it might look like